core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -342,6 +320,87 @@ class ExecutorSuite extends SparkFunSuite
     }
   }
 
+  test("Send task executor metrics in DirectTaskResult") {
+    // Run a successful, trivial result task
+    // We need to ensure, however, that executor metrics are polled after the task is started
+    // so this requires some coordination using ExecutorSuiteHelper.
+    val conf = new SparkConf().setMaster("local").setAppName("executor suite test")
+    sc = new SparkContext(conf)
+    val serializer = SparkEnv.get.closureSerializer.newInstance()
+    ExecutorSuiteHelper.latches = new ExecutorSuiteHelper
+    val resultFunc =
+      (context: TaskContext, itr: Iterator[Int]) => {
+        ExecutorSuiteHelper.latches.latch1.await(300, TimeUnit.MILLISECONDS)
```

Comment:
please increase all of the timeouts added here, to at least 5 seconds, preferably 10 -- its not uncommon for there to be a long pause when running these tests on jenkins, which then leads to flaky tests

Reply:
Will do.